pkg/core/mail/mask.go

```diff
@@ -0,0 +1,42 @@
+package mail
+
+import (
+	gomail "net/mail"
+	"strings"
+)
+
+// MaskAddress masks the given email address, ignoring name.
+func MaskAddress(s string) string {
+	a, err := gomail.ParseAddress(s)
+	if err != nil {
+		return ""
+	}
+
+	// Copied from stdlib
+	// https://golang.org/src/net/mail/message.go?s=5217:5250#L172
+	at := strings.LastIndex(a.Address, "@")
+	var local, domain string
+	if at < 0 {
+		local = a.Address
+	} else {
+		local, domain = a.Address[:at], a.Address[at+1:]
+	}
+	// Copied from stdlib
+
+	runes := []rune(local)
+	length := len(runes)
+	halfLength := length / 2
+
+	var buf strings.Builder
+	for i := 0; i < length; i++ {
+		if i < halfLength {
+			buf.WriteRune(runes[i])
+		} else {
+			buf.WriteRune('*')
+		}
+	}
+	buf.WriteRune('@')
```

Comment:
https://golang.org/src/net/mail/message.go?s=5217:5250#L216

It writes `@` unconditionally. Which part are you referring to?

Reply:
I was thinking if this function receive `"test"`, without the '@' character, it should returns `"te**"`, even if it may not be a valid email address. However after looking into the details, seems this function rejects invalid email address by returning empty string, and the passed in email should be valid email already (read from database). So just ignore my comment.